plugins/badges-backend/config.d.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /**
+   * Define all badges.
+   *
+   * The `label` and `message` fields may use templating to support
+   * dynamic content, based on context. Use same syntax as for
+   * javascript template literals, but with `_` instead of `$`,
+   * e.g. `_{context.variable.name}`.
```

Comment:
The Shields.io spec indicates that the left side (your label in this context) should not be dynamic?

https://github.com/badges/shields/blob/master/spec/SPECIFICATION.md

I would potentially suggest against this, even if it provides more flexibility to the person making the badge, it veers away from the industry standard (read: "expectation") users may have for badges.

Open to counter opinions on this!

Reply:
Good point. I think it would suffice to make this clear in the readme/docs how the label ought to be employed. The "dynamic" nature, could be rather static, if you put like the entity kind in there (or `app.title`), or some such.. So it depends on the context how it is used.